src/backend/cuda/kernel/canny.hpp

```diff
@@ -161,196 +44,28 @@ void nonMaxSuppression(Param<T> output, CParam<T> magnitude, CParam<T> dx,
-        bool _ea = outMem[j][i + 1] == WEAK && VALID_BLOCK_IDX(j, i + 1);
-        bool _se =
-            outMem[j + 1][i + 1] == WEAK && VALID_BLOCK_IDX(j + 1, i + 1);
-        bool _so = outMem[j + 1][i] == WEAK && VALID_BLOCK_IDX(j + 1, i);
-        bool _sw =
-            outMem[j + 1][i - 1] == WEAK && VALID_BLOCK_IDX(j + 1, i - 1);
-        bool _we = outMem[j][i - 1] == WEAK && VALID_BLOCK_IDX(j, i - 1);
-
-        bool hasWeakNeighbour =
-            _nw || _no || _ne || _ea || _se || _so || _sw || _we;
-
-        continueIter = __syncthreads_or(cu == STRONG && hasWeakNeighbour);
-    };
-
-    // Check if any 1-pixel border ring
-    // has weak pixels with strong candidates
-    // within the main region, then increment hasChanged.
-    int cu = outMem[j][i];
-    int nw = outMem[j - 1][i - 1];
-    int no = outMem[j - 1][i];
-    int ne = outMem[j - 1][i + 1];
-    int ea = outMem[j][i + 1];
-    int se = outMem[j + 1][i + 1];
-    int so = outMem[j + 1][i];
-    int sw = outMem[j + 1][i - 1];
-    int we = outMem[j][i - 1];
-
-    bool hasWeakNeighbour = nw == WEAK || no == WEAK || ne == WEAK ||
-                            ea == WEAK || se == WEAK || so == WEAK ||
-                            sw == WEAK || we == WEAK;
-
-    if (__syncthreads_or(cu == STRONG && hasWeakNeighbour) && lx == 0 &&
-        ly == 0)
-        atomicAdd(&hasChanged, 1);
-
-    // Update output with shared memory result
-    if (gx < (output.dims[0] - 2) && gy < (output.dims[1] - 2))
-        oPtr[lIdx(gx, gy, output.strides[0], output.strides[1])] = outMem[j][i];
-}
-
-template<typename T>
-static __global__ void suppressLeftOverKernel(Param<T> output, unsigned nBBS0,
-                                              unsigned nBBS1) {
-    // batch offsets for 3rd and 4th dimension
-    const unsigned b2 = blockIdx.x / nBBS0;
-    const unsigned b3 = blockIdx.y / nBBS1;
-
-    // global indices
-    const int gx = blockDim.x * (blockIdx.x - b2 * nBBS0) + threadIdx.x;
-    const int gy = blockDim.y * (blockIdx.y - b3 * nBBS1) + threadIdx.y;
-
-    // Offset input and output pointers to second pixel of second coloumn/row
-    // to skip the border
-    T* oPtr = output.ptr + (b2 * output.strides[2] + b3 * output.strides[3]) +
-              output.strides[1] + 1;
-
-    if (gx < (output.dims[0] - 2) && gy < (output.dims[1] - 2)) {
-        int idx = lIdx(gx, gy, output.strides[0], output.strides[1]);
-        T val   = oPtr[idx];
-        if (val == WEAK) oPtr[idx] = NOEDGE;
-    }
-}
-
 template<typename T>
 void edgeTrackingHysteresis(Param<T> output, CParam<T> strong, CParam<T> weak) {
+    static const std::string source(canny_cuh, canny_cuh_len);
+
+    auto init =
+        getKernel("cuda::initEdgeOut", source, {TemplateTypename<T>()},
+                  {DefineValue(STRONG), DefineValue(WEAK), DefineValue(NOEDGE),
+                   DefineValue(THREADS_X), DefineValue(THREADS_Y)});
+    auto trace =
```

Comment:
Make your kernel variable names match the kernel names inside the nvrtc files. It makes it easier to grep for names that way.